woocommerce/woocommerce-ios
# [Mobile Payments] Use a single object for token and location, update its context as siteID, gateway change

Closes: #5912

### Description
- Currently we pass a token provider to the `start` method of the `CardReaderService`
- That token provider is passed to the Terminal SDK so that it can request a token when attempting to connect a reader
- Unfortunately, the Terminal SDK will continue to call that object once set
- So, we instead have CardPresentPaymentStore instantiate an object that can accept changes to the siteID and backend and update it each time we start discovery. That way when Stripe calls into it, it has the correct context.

### Testing instructions
- Enable the Stripe IPP experiment if you haven't already
- Connect to a card reader for a store using WCPay as its payments backend
- Switch to a store using Stripe as its payments backend
- Ensure you can connect to a card reader there too

### Screenshots
N/A

---
